runtimes/runtimes/operational-telemetry/aws-metric-exporter.ts

```diff
@@ -0,0 +1,172 @@
+import { Sha256 } from '@aws-crypto/sha256-js'
+import {
+    CognitoIdentityClient,
+    Credentials,
+    GetCredentialsForIdentityCommand,
+    GetIdCommand,
+} from '@aws-sdk/client-cognito-identity'
+import { ExportResult, ExportResultCode } from '@opentelemetry/core'
+import { MetricData, PushMetricExporter, ResourceMetrics, ScopeMetrics } from '@opentelemetry/sdk-metrics'
+import { HttpRequest } from '@smithy/protocol-http'
+import { SignatureV4 } from '@smithy/signature-v4'
+import axios from 'axios'
+import { OperationalMetric, OperationalTelemetry } from './operational-telemetry'
+import { Resource } from '@opentelemetry/resources'
+
+export class AWSMetricExporter implements PushMetricExporter {
+    private readonly endpoint: string
+    private readonly region: string
+    private readonly poolId: string
+    private readonly telemetryService: OperationalTelemetry
+
+    private isShutdown = false
+
+    private credentials: Credentials | null = null
+    private credentialsLastFetched: Date | null = null
+    private readonly CREDENTIALS_EXPIRATION_TIME_MS = 60 * 60 * 1000 // 60 minutes, default for cognito crednetials
+    private readonly CREDENTIALS_BUFFER_TIME_MS = 1 * 60 * 1000 // 1 min
+
+    constructor(endpoint: string, region: string, poolId: string, telemetryService: OperationalTelemetry) {
+        this.endpoint = endpoint
+        this.region = region
+        this.poolId = poolId
+        this.telemetryService = telemetryService
+    }
+
+    async export(metrics: ResourceMetrics, resultCallback: (result: ExportResult) => void): Promise<void> {
+        if (this.isShutdown) {
+            setImmediate(resultCallback, { code: ExportResultCode.FAILED })
+            return
+        }
+
+        try {
+            const operationalMetrics = this.extractOperationalMetrics(metrics)
+            await this.refreshCognitoCredentials(this.region, this.poolId)
+            await this.sendOperationalMetrics(operationalMetrics)
+            resultCallback({ code: ExportResultCode.SUCCESS })
+        } catch (error) {
+            console.log('Failed to export metrics', error)
+            resultCallback({ code: ExportResultCode.FAILED })
+            return
+        }
+    }
+
+    forceFlush(): Promise<void> {
+        return Promise.resolve()
+    }
+
+    shutdown(): Promise<void> {
+        this.isShutdown = true
+        return Promise.resolve()
+    }
+
+    async refreshCognitoCredentials(region: string, poolId: string): Promise<Credentials> {
+        if (this.credentials && this.credentialsLastFetched) {
+            const now = new Date()
+            const timeSinceLastFetch = now.getTime() - this.credentialsLastFetched.getTime()
+
+            if (timeSinceLastFetch < this.CREDENTIALS_EXPIRATION_TIME_MS - this.CREDENTIALS_BUFFER_TIME_MS) {
```

Comment:
Shouldn't it be ">"? Can we have tests for such kind of functionality added?

Reply:
I will add some unit tests after the logic is done. I tested it manually.